### ISSUE TITLE
Settle on a single semver library

### ISSUE THREAD

**Issue by timstclair** [kubernetes/kubernetes#23854]
We currently depend on 2 semver libraries:
- https://github.com/coreos/go-semver
- https://github.com/blang/semver

``` console
$ git grep "github.com/.*semver" pkg test
pkg/kubelet/dockertools/manager.go:     "github.com/coreos/go-semver/semver"
pkg/kubelet/rkt/version.go:     "github.com/coreos/go-semver/semver"
pkg/proxy/iptables/proxier.go:  "github.com/coreos/go-semver/semver"
pkg/util/iptables/iptables.go:  "github.com/coreos/go-semver/semver"
pkg/version/semver.go:  "github.com/blang/semver"
test/e2e/util.go:       "github.com/blang/semver"
```

I'm not sure which library is better, but we should probably agree on one for consistency.

**Comment by danwinship** [comment]
See also #32401; in pkg/proxy/iptables/proxier.go and pkg/util/iptables/iptables.go, we shouldn't be doing a semver compare anyway, because iptables version numbers aren't always syntactically Semantic (eg "1.4.19.1" which is illegal because semvers can't have nano versions).

**Comment by danwinship** [comment]
more-or-less-fixed by #35436; now no part of kubernetes itself uses either library, but it still depends on both of them indirectly (blang/semver via cadvisor and coreos/go-semver via etcd) so we still can't get rid of either of them
